pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -496,8 +499,16 @@ public String getForwardIndexFileName(String column, String segmentVersion) {
     // if (!SegmentVersion.v1.toString().equalsIgnoreCase(segmentVersion)) {
     // fileNameBuilder.append("_").append(segmentVersion);
     // }
-    if (columnMetadata.isSingleValue()) {
-      if (columnMetadata.isSorted()) {
+    
+    boolean singleValue = true;
+    boolean sorted = true;
+    if(columnMetadata!=null){
+      singleValue = columnMetadata.isSingleValue();
+      sorted = columnMetadata.isSorted();
+    }
```

Comment:
When missing column metadata, always assume single value? How to handle multi value missing column?